infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutAlignmentDao.kt

```diff
@@ -48,10 +49,15 @@ class LayoutAlignmentDao(
 
     fun fetchVersions() = fetchRowVersions<LayoutAlignment>(LAYOUT_ALIGNMENT)
 
+    @Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
     fun fetch(alignmentVersion: RowVersion<LayoutAlignment>): LayoutAlignment =
         if (cacheEnabled) alignmentsCache.get(alignmentVersion, ::fetchInternal)
         else fetchInternal(alignmentVersion)
 
+    @Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
+    fun fetchMany(versions: List<RowVersion<LayoutAlignment>>): Map<RowVersion<LayoutAlignment>, LayoutAlignment> =
```

Comment:
Tämähän siis hakee kaikki edelleen yksittäin koska id+versio multifetch vaatisi monimutkaisemman sql:n. Pointti on se että saadaan kuitenkin ylemmälle tasolle truo transaktioraja - tämän itse kutsuma fetch ignoraa fetch-funktion annotaation. Tämä tiputtaa annotaation aiheuttamaa extrahäröilyä melko oleellisesti.